ansible/roles/test/files/ptftests/advanced-reboot.py

```diff
@@ -350,6 +350,18 @@ def populate_fail_info(self, fails):
                 self.fails[key] = set()
             self.fails[key] |= fails[key]
 
+    def get_preboot_info(self):
+        msg = 'Preboot oper: %s ' % self.preboot_oper
+        if ':' in self.preboot_oper:
+            temp = self.preboot_oper.split(':')
+            msg = 'Preboot oper: %s ' % temp[0]
+            if len(temp) > 2:
+                msg += 'Number of sad path VMs: %s Lag member down in a portchannel: %s' % (temp[-2], temp[-1])
```

Comment:
It would be helpful to add an example of input string here for reference. Also the code has an implicit decision that only the last 2 values were used. You should add a comment to explain the choice.

Reply:
added more comments